src/figuresClasses.ts

```diff
@@ -1,11 +1,81 @@
-export interface Figure {}
+type Colors = 'red' | 'green' | 'blue';
+type Shapes = `triangle` | `circle` | `rectangle`;
 
-export class Triangle implements Figure {}
+export interface Figure {
+  readonly shape: Shapes;
+  color: Colors;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  readonly shape: Shapes = `triangle`;
+
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
```

Comment:
[CODE KNOWLEDGE] - When you declare a constructor parameter with an access modifier, TypeScript implicitly creates a property on the class with the same name as the parameter. There's no need to declare the properties 'color', 'a', 'b', and 'c' manually.